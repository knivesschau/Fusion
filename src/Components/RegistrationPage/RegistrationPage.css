/* mobile styles first, min 320px */
.Registration_Page {
    overflow-y: hidden;
    font-size: 1.10rem;
    text-align: center;
    background-color: rgb(190, 190, 190, .8);
    border: 1px solid grey;
    border-radius: 2em;
    box-shadow: inset -2px -2px 5px 0 black;
    margin: 1.5em;
    margin-top: 1.75em;
    padding: 1em;
}

.Register_Form {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-top: -1.75em;
}

#register-tag {
    font-size: 1.40rem;
    color: white; 
    text-shadow: 3px 2px 2px black;
    margin-top: -.25em;
}

#new-user, #new-password, #confirm-password {
    font-size: 1.25rem;
    font-weight: 500;
    margin-bottom: .25em;
}

#reg-user, #reg-password, #confirm-pass {
    transition: border .3s ease-out; 
    padding: .25em;
    width: 70%;
}

#reg-user:focus, #reg-password:focus, #confirm-pass:focus {
    border: 10px solid #881600;
    border-style: none none none solid;
    background-color: white;
}

#register-new {
    cursor: pointer;
    display: inline-block;
    box-sizing: border-box;
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-size: 1rem;
    font-weight: bold;
    margin-left: auto;
    margin-right: auto;
    margin-top: 1.25em;
    margin-bottom: .75em;
    padding: 7px 25px;
    border: 1px solid black;
    border-radius: .75em;
    transition: all .15s;
}

#register-new:hover {
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#register-new:active {
    background-color: #881600;
}

#cutlery-icon {
    height: 45%;
    width: 45%;
}

/* tablet and desktop styles */
@media only screen and (min-width: 768px) {
    .Registration_Page {
        margin-left: 5em;
        margin-right: 5em;
        margin-top: 3em;
        border-radius: 5em;
    }

    #register-tag {
        font-size: 2.5rem;
        margin-bottom: .5em;
    }

    #new-user, #new-password, #confirm-password, #register-new {
        font-size: 1.5rem;
    }

    #register-new {
        margin-bottom: .5em;
        padding-top: .45em;
        padding-bottom: .45em;
    }

    #reg-user, #reg-password, #confirm-pass {
        width: 60%;
        font-size: 1.25rem;
    }

    #cutlery-icon {
        height: 30%;
        width: 30%;
    }
}

@media only screen and (min-width: 1200px) {
    .Registration_Page {
        margin-left: 15em;
        margin-right: 15em;
    }

    #register-tag {
        font-size: 3.5rem;
    }

    #new-user, #new-password, #confirm-password, #register-new {
        font-size: 2rem;
    }

    #reg-user, #reg-password, #confirm-pass {
        width: 50%;
        font-size: 1.5rem;
    }

    #cutlery-icon {
        height: 25%;
        width: 25%;
    }
}

@media only screen and (min-width: 1600px) {
    .Registration_Page {
        margin-left: 25em;
        margin-right: 25em;
        margin-top: 5em;
    }

    #cutlery-icon {
        height: 15%;
        width: 15%;
    }
}

@media only screen and (min-width: 2000px) {
    .Registration_Page {
        margin-left: 40em;
        margin-right: 40em;
    }
}