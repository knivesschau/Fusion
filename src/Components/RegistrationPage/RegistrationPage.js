import React, {Component} from 'react';
import './RegistrationPage.css';

export default class RegistrationPage extends Component {
    render() {
        return (
            <section className="Registration_Page">

                <h3>Let's Get Cooking! (Will Be Included in Final Version)</h3>

                <form className="Register_Form">

                    <label htmlFor="username">
                        <p id="new-user">Username:</p>
                        <input type="text" id="reg-user" name="register_username"/>
                    </label>

                    <label htmlFor="password">
                        <p id="new-password">Password:</p>
                        <input type="password" id="reg-password" name="register_password"/>
                    </label>

                    <label htmlFor="confirm-password">
                        <p id="confirm-pass">Re-Type Password:</p>
                        <input type="password" id="confirm-password" name="confirm_pass"/>
                    </label>
                    
                    <button type="submit" id="register-new">Register</button>
                    
                </form>

            </section>
        );
    }
}