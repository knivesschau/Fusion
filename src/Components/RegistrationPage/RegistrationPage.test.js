import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationPage from './RegistrationPage';

describe ('RegistrationPage Component', () => {
    it ('Renders the complete RegistrationPage Component and all nested Components', () => {
        const wrapper = shallow(<RegistrationPage/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});