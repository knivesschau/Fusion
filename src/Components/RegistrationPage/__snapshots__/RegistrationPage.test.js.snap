// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage Component Renders the complete RegistrationPage Component and all nested Components 1`] = `
<section
  className="Registration_Page"
>
  <img
    alt="cutlery"
    id="cutlery-icon"
    src="cutlery.png"
  />
  <h3
    id="register-tag"
  >
    Let's Get Cooking!
  </h3>
  <form
    className="Register_Form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      <p
        id="new-user"
      >
        Username:
      </p>
      <input
        id="reg-user"
        name="register_username"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="password"
    >
      <p
        id="new-password"
      >
        Password:
      </p>
      <input
        id="reg-password"
        name="register_password"
        onChange={[Function]}
        type="password"
      />
    </label>
    <label
      htmlFor="confirm-password"
    >
      <p
        id="confirm-password"
      >
        Re-Type Password:
      </p>
      <input
        id="confirm-pass"
        name="confirm_pass"
        onChange={[Function]}
        type="password"
      />
    </label>
    <button
      id="register-new"
      type="submit"
    >
      Register
    </button>
  </form>
  <ErrorValidation />
  <ErrorValidation />
  <ErrorValidation />
</section>
`;
